src/mame/drivers/swtpc8212.cpp

```diff
@@ -22,82 +60,359 @@ class swtpc8212_state : public driver_device
 	swtpc8212_state(const machine_config &mconfig, device_type type, const char *tag)
 		: driver_device(mconfig, type, tag)
 		, m_maincpu(*this, "maincpu")
-		, m_chargen(*this, "chargen")
+		, m_pia0(*this, "pia0")
+		, m_pia1(*this, "pia1")
+		, m_uart(*this, "uart")
+		, m_crtc(*this, "crtc")
+		, m_chargen1(*this, "chargen1")
+		, m_chargen2(*this, "chargen2")
+		, m_video_ram(*this, "videoram")
+		, m_baud_rate(*this, "baud_rate")
+		, m_no_parity(*this, "no_parity")
+		, m_parity_select(*this, "parity_select")
+		, m_jumper_a(*this, "jumper_a")
+		, m_jumper_b(*this, "jumper_b")
+		, m_mode_switch(*this, "mode_switch")
+		, m_bell_timer(nullptr)
+		, m_beeper(*this, "beeper")
+		, m_printer(*this, "printer")
 	{
 	}
 
 	void swtpc8212(machine_config &mconfig);
 
-private:
-	MC6845_UPDATE_ROW(update_row);
+	enum
+	{
+		BELL_TIMER_ID = 1
+	};
 
+private:
 	void mem_map(address_map &map);
+	virtual void machine_reset() override;
+	virtual void machine_start() override;
+	virtual void device_timer(emu_timer &timer, device_timer_id id, int param, void *ptr) override;
```

Comment:
Please don't make overridden base members private - it makes it harder to make derived state classes for related machines.